modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/ConfigurationModelParser.java

```diff
@@ -101,7 +102,8 @@ public interface ConfigurationModelParser extends StereotypeModelParser, Additio
   Optional<DisplayModel> getDisplayModel();
 
   /**
-   * @return a {@link MuleVersion} representing the minimum mule version this component can run on
+   * @return a {@link MinMuleVersionResult} that contains the minimum mule version this component can run on and the reason why
+   *         that version was assigned.
    */
-  Optional<MuleVersion> getMinMuleVersion();
```

Comment:
the reasoning for this to be optional is decoupling the parser from the default. For instance, if the parser doesn't determine any specific version, the delegate can default to the extension's MMV without the need to pass it all around. What's the justification for this change? Why is it being done in a PR that's supposed to be about logging?

Reply:
Good point. I was thinking that the parser would always return a result and would handle the default case too. But maybe what you say makes more sense. I'm gonna return it to an optional then.